spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -401,6 +401,46 @@ public void addFilteredPartitionsToPartitioned2() {
         sql("SELECT id, name, dept, subdept FROM %s ORDER BY id", tableName));
   }
 
+  @Test
+  public void addFilteredPartitionsToPartitionedWithNullValue() {
+    createCompositePartitionedTableWithNullValueInPartitionColumn("parquet");
+
+    String createIceberg =
+        "CREATE TABLE %s (id Integer, name String, dept String, subdept String) USING iceberg " +
+            "PARTITIONED BY (id, dept)";
+
+    sql(createIceberg, tableName);
+
+    Object result = scalarSql("CALL %s.system.add_files('%s', '`parquet`.`%s`', map('id', 1))",
+        catalogName, tableName, fileTableDir.getAbsolutePath());
+
+    Assert.assertEquals(2L, result);
+
+    assertEquals("Iceberg table contains correct data",
+        sql("SELECT id, name, dept, subdept FROM %s WHERE id = 1 ORDER BY id", sourceTableName),
+        sql("SELECT id, name, dept, subdept FROM %s ORDER BY id", tableName));
+  }
+
+  @Test
+  public void addFilteredPartitionsToPartitionedWithNullValue2() {
```

Comment:
2? Is there a better name to distinguish between this one and the test above?

Reply:
This is the naming format from `addFilteredPartitionsToPartitioned`, but I'll see about a better name.

Reply:
I updated the names of both tests.

There's `addFilteredPartitionsToPartitionedWithNullValueFilteringOnDept` and `addFilteredPartitionsToPartitionedWithNullValueFilteringOnId`.

Quite a mouthful, but it's better than just adding 2 at the end I suppose.